components/raftstore/src/store/peer.rs

```diff
@@ -1320,11 +1385,7 @@ impl Peer {
         self.add_ready_metric(&ready, &mut ctx.raft_metrics.ready);
 
         if !ready.committed_entries.as_ref().map_or(true, Vec::is_empty) {
-            // We must renew current_time because this value may be created a long time ago.
-            // If we do not renew it, this time may be smaller than propose_time of a command,
-            // which was proposed in another thread while this thread receives its AppendEntriesResponse
-            //  and is ready to calculate its commit-log-duration.
-            ctx.current_time.replace(monotonic_raw_now());
```

Comment:
This should not happen because `current_time` is shared by FSMs within same batch, and will be cleared at `PollHandler::end`. For the same FSM, this `current_time` will be monotonic, because a FSM will not be handled in diffrent batch at the same time. cc@Little-Wallace

Reply:
/cc @Little-Wallace

Reply:
No, if you read the note I written, you will know why I changed it. 
current_time can not be shared by FSMs within same batch. Because this time may be a long time ago. If this FSM was handled by other thread, this current_time may be less than current_time of the other thread.

Reply:
For example:
thread A:     end() -->       begin() --> FSM1 ---> FSM2 --> end
thread B:     begin()  --> FSM2 ----> FSM3 ------> FSM4 ---------> FSM1 --> end  
So current-time of  thread B may be less than current-time of thread A.
And you will make commit-time of FSM1 less than start-commit-time of it.

Reply:
Yes, I have check out the latest  code of batch system, looks like https://github.com/tikv/tikv/pull/6991 had changed the way to fetch FSM, and situation you mentioned may happen.